code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -69,14 +69,19 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 			cell = null
 			update_icons()
 			diag_hud_set_borgcell()
+	else if(!opened)
+		..()
+
+/mob/living/silicon/robot/attack_hulk(mob/living/carbon/human/user)
+	. = ..()
+	if(!.)
+		return
+	spark_system.start()
+	INVOKE_ASYNC(src, .proc/hulk_punched, user)
 
-	if(!opened)
-		if(..()) // hulk attack
-			spark_system.start()
-			spawn(0)
-				step_away(src,user,15)
-				sleep(3)
-				step_away(src,user,15)
+/mob/living/silicon/robot/proc/hulk_punched(mob/living/carbon/human/user)
+	step_away(src, user, 15)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/_step_away, src, user, 15), 3)
```

Comment:
I would probably also replace user with get_turf(user) here since it doesn't really make sense if the direction of the push changes if the attacker moves post attack.

Reply:
Fixed.